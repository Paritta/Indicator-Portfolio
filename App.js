import React, { Component } from 'react';
import { Text, View, Button } from 'react-native';

//router load
import { Router, Scene } from 'react-native-router-flux'

//Screen Component load
import HomeScreen from './src/Screen/HomeScreen'
import AircornScreen from './src/Screen/AircornScreen'
import InputScreen from './src/Screen/InputScreen'


export default class App extends Component {
      
  render() {
    return (
      <Router>
        <Scene key="root">
          <Scene
            key="homescreen"
            component={HomeScreen}
            title="Home"
            initial
          />
          <Scene
            key="aircornscreen"
            component={AircornScreen}
            title="Aircorn"
          />
          <Scene
            key="inputscreen"
            component={InputScreen}
            title="Input"
          />
        </Scene>
      </Router>
    );
  }
}