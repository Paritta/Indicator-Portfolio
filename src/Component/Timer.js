import React, { Component } from 'react';
import { Text, View, StyleSheet } from 'react-native';
import styled from 'styled-components/native';
import { Button } from 'react-native-elements'
import Indicator from './Indicator'

export default class Timer extends Component {
 
    constructor( props ) {
        super( props );

        this.state = {
            timer: null,
            hour: '0',
            counter: '00',
            miliseconds: '00',
            Indicator: '0',
            startDisabled: true,
            stopDisabled: false,
        }

        this.onButtonStart = this.onButtonStart.bind(this);
        this.onButtonStop = this.onButtonStop.bind(this);
        this.onButtonClear = this.onButtonClear.bind(this);
        this.start = this.start.bind(this);
    }
      componentDidMount() {
        this.start();  
      }
      
      componentWillMount() {
        clearInterval(this.state.timer);
      }

      start() {
        var self = this;
        let timer = setInterval(()=> {
          var num = (Number(this.state.miliseconds) + 1).toString(),
              count = this.state.counter,
              hour = this.state.hour;
          var Indicator = (Number(this.state.miliseconds) + 1).toString();

          if( Number(this.state.miliseconds) == 59 ) {
              count = (Number(this.state.counter) + 1).toString();
              num = '00';
          }

          if( Number(this.state.counter) == 59 ) {
              hour = (Number(this.state.hour) + 1).toString();
              count = '00';
          }
          
          self.setState({
              counter: count.length == 1 ? '0'+count : count,
              miliseconds: num.length == 1 ? '0'+num : num,
              Indicator: Indicator,
              hour: hour
          });
        }, 1000);
        this.setState({timer});
      }

      onButtonStart() {
          this.start();
          this.setState({startDisabled: true, stopDisabled: false});
      }

      onButtonStop() {
          clearInterval(this.state.timer);
          this.setState({startDisabled: false, stopDisabled: true});
      }

      onButtonClear() {
          this.setState({
            timer: null,
            counter: '00',
            miliseconds: '00'
          });
      }
      
  render() {
    return (
      <ContainerView>
        <Indicator Indicator={this.state.Indicator}/>
        <CounterText>{this.state.hour} : {this.state.counter} : {this.state.miliseconds}</CounterText>
            <Button
              disabled={this.state.startDisabled}
              onPress={this.onButtonStart}
              title="Start"
              buttonStyle={styles.someButtonStyle}
              icon={{name: 'grain'}}
            />
            <Button
              disabled={this.state.stopDisabled}
              onPress={this.onButtonStop}
              title="Stop"
              buttonStyle={styles.someButtonStyle}
              icon={{name: 'grain'}}
            />
            <Button
              disabled={this.state.startDisabled}
              onPress={this.onButtonClear}
              title="Clear"
              buttonStyle={styles.someButtonStyle}
              icon={{name: 'grain'}}
            />
      </ContainerView>
    );
  }
}

const ContainerView = styled.View`
  flex: 1;
  justify-content: center;
  align-items: center;
  opacity: 0.6;
`;

const CounterText = styled.Text`
    color: white;
    font-size: 30;
    text-align: center;
    height: 60;
    margin: 10px;
`

var styles = StyleSheet.create({
  someButtonStyle: {
    padding: 20,
    height:20,
    width: 350,
    backgroundColor: 'mediumspringgreen',
    marginTop:5
  },
});