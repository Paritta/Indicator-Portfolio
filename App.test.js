import React from 'react';
import App from './App';
import Timer from './src/Component/Timer'
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  let rendered = renderer.create(<Timer />).toJSON();
  expect(rendered).toMatchSnapshot();


  rendered = renderer.create(<Timer />).toJSON();
  expect(rendered).toMatchSnapshot();
});

// it('renders without crashing', () => {
//   const rendered = renderer.create(<App />).toJSON();
//   expect(rendered).toBeTruthy();
// });
