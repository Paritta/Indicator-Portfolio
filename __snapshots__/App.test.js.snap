// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "opacity": 0.6,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "white",
          "borderRadius": 140,
          "borderStyle": "solid",
          "borderWidth": 5,
          "height": 280,
          "justifyContent": "center",
          "marginBottom": 10,
          "width": 280,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      Indicator
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 30,
          "height": 60,
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    00
     : 
    00
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "mediumspringgreen",
              "height": 20,
              "marginTop": 5,
              "padding": 20,
              "width": 350,
            },
            Object {
              "backgroundColor": "#dadee0",
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 18,
              },
              Array [
                Object {
                  "marginRight": 10,
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Start
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "mediumspringgreen",
              "height": 20,
              "marginTop": 5,
              "padding": 20,
              "width": 350,
            },
            false,
            false,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 18,
              },
              Array [
                Object {
                  "marginRight": 10,
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Stop
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "mediumspringgreen",
              "height": 20,
              "marginTop": 5,
              "padding": 20,
              "width": 350,
            },
            Object {
              "backgroundColor": "#dadee0",
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 18,
              },
              Array [
                Object {
                  "marginRight": 10,
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Clear
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders without crashing 2`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "opacity": 0.6,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "white",
          "borderRadius": 140,
          "borderStyle": "solid",
          "borderWidth": 5,
          "height": 280,
          "justifyContent": "center",
          "marginBottom": 10,
          "width": 280,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      Indicator
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 30,
          "height": 60,
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    00
     : 
    00
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "mediumspringgreen",
              "height": 20,
              "marginTop": 5,
              "padding": 20,
              "width": 350,
            },
            Object {
              "backgroundColor": "#dadee0",
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 18,
              },
              Array [
                Object {
                  "marginRight": 10,
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Start
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "mediumspringgreen",
              "height": 20,
              "marginTop": 5,
              "padding": 20,
              "width": 350,
            },
            false,
            false,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 18,
              },
              Array [
                Object {
                  "marginRight": 10,
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Stop
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "mediumspringgreen",
              "height": 20,
              "marginTop": 5,
              "padding": 20,
              "width": 350,
            },
            Object {
              "backgroundColor": "#dadee0",
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 18,
              },
              Array [
                Object {
                  "marginRight": 10,
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Clear
        </Text>
      </View>
    </View>
  </View>
</View>
`;
